docs/fundamentals/custom-resource-urls.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Custom resource URLs
+description: Learn how to create custom URLs for .NET Aspire resources.
+ms.date: 04/08/2025
+ms.topic: how-to
+---
+
+# Custom resource URLs
+
+.NET Aspire resources that expose endpoints only configure host and port values. However, there might be situations where you want to access a specific route of an exposed endpoint. The host and port are unknown until run time. In these cases, you can use custom resource URLs to define specific routes on a configured endpoint, which is convenient for accessing resources from the [dashboard](dashboard/overview.md).
+
+## Default endpoint behavior
+
+By default, as described in the [Networking inner loop](networking-overview.md#networking-in-the-inner-loop) article, .NET Aspire relies on existing configurations such as Kestrel or launch profiles to determine the host and port of a resource for a configured endpoint.
```

Comment:
Can we clarify that this is only true for ASPNET Core project resources?

Reply:
Ah, so it's not applicable to executable or container resources?

Reply:
Only Project resources have their Kestrel configuration and launch profiles inspected for endpoints/URLs.